src/presentation/http/types/HttpRequestContext.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Represents the HTTP request context with data, passed to the route handler from middlewares
+ */
+export interface HttpRequestContext<AuthPayload> {
```

Comment:
i think you can get rid of generic here and use regular import

Reply:
done